main.nf

```diff
@@ -277,99 +282,267 @@ if( params.bwa_index && (params.aligner == 'bwa' | params.bwamem)){
             .map { row -> [ row[0], [ file( row[1][0] ), file( row[1][1] ) ] ] }
-            .ifEmpty { exit 1, "params.readPaths or params.bams was empty - no input files supplied!" }
-            .into { ch_read_files_clip; ch_read_files_fastqc; ch_read_files_complexity_filter_poly_g; ch_read_unmap }
-            ch_bam_to_fastq_convert = Channel.empty()
+            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied!" }
+            .into { ch_input_for_skipconvertbam; ch_input_for_convertbam; ch_input_for_indexbam }
     } else {
         Channel
             .from( params.readPaths )
+            .filter { it =~/.*.bam/ }
+            .ifEmpty { exit 1, "Your specified BAM read files did not end in: '.bam' " }
             .map { row -> [ file( row )  ] }
-            .ifEmpty { exit 1, "params.readPaths or params.bams was empty - no input files supplied!" }
+            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied!" }
             .dump()
-            .set { ch_bam_to_fastq_convert }
+            .into { ch_input_for_skipconvertbam; ch_input_for_convertbam; ch_input_for_indexbam }
 
-            //Set up clean channels
-            ch_read_files_fastqc = Channel.empty()
-            ch_read_files_complexity_filter_poly_g = Channel.empty()
-            ch_read_files_clip = Channel.empty()
-            ch_read_unmap = Channel.empty()
     }
 } else if (!params.bam){
      Channel
-        .fromFilePairs( params.reads, size: params.singleEnd ? 1 : 2 )
-        .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}\nNB: Path needs" +
-            "to be enclosed in quotes!\nNB: Path requires at least one * wildcard!\nIf this is single-end data, please specify --singleEnd on the command line." }
-        .into { ch_read_files_clip; ch_read_files_fastqc; ch_read_files_complexity_filter_poly_g; ch_read_unmap }
-        ch_bam_to_fastq_convert = Channel.empty()
+        .fromFilePairs( params.reads, size: params.single_end ? 1 : 2 )
+        .filter { it =~/.*.fastq.gz|.*.fq.gz|.*.fastq|.*.fq/ }
+        .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}\nNB: Path needs " +
+            "to be enclosed in quotes!\nNB: Path requires at least one * wildcard!\nValid input file types: .fastq.gz', '.fq.gz', '.fastq', or '.fq'\nIf this is single-end data, please specify --single_end on the command line." }
+        .into { ch_input_for_skipconvertbam; ch_input_for_convertbam; ch_input_for_indexbam  }
+
 } else {
      Channel
         .fromPath( params.reads )
+        .filter { it =~/.*.bam/ }
         .map { row -> [  file( row )  ] }
-        .ifEmpty { exit 1, "Cannot find any bam file matching: ${params.reads}\nNB: Path needs" +
+        .ifEmpty { exit 1, "Cannot find any bam file matching: ${params.reads}\nValid input file types: .fastq.gz', '.fq.gz', '.fastq', or '.fq'\nNB: Path needs " +
```

Comment:
Error message not matching input filtering. Should be something like `.ifEmpty { exit 1, "Your specified BAM read files did not end in: '.bam' " }`